src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -14,18 +14,44 @@ interface DropdownMenuItemProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   className?: string;
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  // To select a certain option in the menu without having to use document.querySelector
+  customRef?: React.RefObject<HTMLDivElement>;
+  onActivate?: (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) => void;
```

Comment:
Perhaps remove the `onClick` event, and only use `onActivate`? Or perhaps only keep `onClick` and call that one if the user activates the button as well? How does a regular `<div>` tag handle clicks and pressing enter? We should do the same.

How does a regular `<button>` or `<div>` work? I think we should be really close to how native HTML works, and do exactly the same. Or at least the same way that `react` handles these elements. I believe they just call the `onClick` event when you press enter.

Reply:
I think it's better to keep the onClick for backward compatibility.  I will change it to call the onClick when user activates the option as well.

Reply:
Actually there is a problem with calling the onClick when pressing enter, one is a keyboard event and one is a mouse event. I would have to forcefully cast it to any to be able to call onClick. Maybe it's fine to keep both.

Reply:
`React.HTMLProps<HTMLDivElement>` doesn't it contain all these props already? `onClick`, 'className` and `onClose` so them being defined is unnecessary.

But what's the difference between `onActivate` and `onClick`? Why do we need both? Shouldn't they do the same.

Reply:
> `React.HTMLProps<HTMLDivElement>` doesn't it contain all these props already? `onClick`, 'className`and`onClose` so them being defined is unnecessary.
> 
Yes, I can remove those. The React.HTMLProps<HTMLDivElement> was added later, that's why those were there before.

> But what's the difference between `onActivate` and `onClick`? Why do we need both? Shouldn't they do the same.
The onActivate is when you activate the menu option, doesn't matter if it's through keyboard or mouse.

Reply:
👍 thanks!

Reply:
https://stackoverflow.com/questions/32659099/making-a-clickable-div-accessible-through-tab-structure/32659785

The accepted answer here makes some good points about creating accessible elements. Perhaps it would be better to use a `<button>` element, not sure...

I don't have enough knowledge about how to make an app accessible.

Reply:
That would've been easier, but using button will break the stylings though...

Reply:
For (dropdown) list, W3.org suggests
`<ul><li></li>...</ul>`
https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html

Reply:
The thing is we are not really creating a dropdown here even though we call it dropdown menu.  I remember there used to be a real dropdown in UIToolkit-components. But the styles did not align with the other dropdowns in Symphony. That's why we had to create a new dropdown menu component. And I think the styles were not intended to work on list or buttons...